### PR TITLE
`Development`: Add JavaDocs to UtilServices and Factories [17]

### PR DESCRIPTION
**Pull request by DominikRemo**
#### General
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context

Improve the readability of UtilServices and Factories by adding JavaDoc to the helper methods

### Description

Added documentation to following files:
- [x] bonus/BonusFactory.java
- [x] competency/CompetencyUtilService.java

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### CODE REVIEW

File: src/test/java/de/tum/in/www1/artemis/bonus/BonusFactory.java [hunk 1]
```diff
@@ -10,13 +10,13 @@
...
+     * Generates a Bonus with the given arguments.
      *
-     * @param bonusStrategy         of bonus
-     * @param weight                of bonus
-     * @param sourceGradingScaleId  of sourceGradingScale of bonus
-     * @param bonusToGradingScaleId of bonusToGradingScale bonus
-     * @return a new Bonus instance associated with the grading scales corresponding to ids bonusToGradingScaleId and bonusToGradingScaleId.
+     * @param bonusStrategy         The bonus strategy of the bonus
+     * @param weight                The weight of the bonus
+     * @param sourceGradingScaleId  The id of the grading scale that is used as the source for the bonus
```
Comment (egekurt123) [inline src/test/java/de/tum/in/www1/artemis/bonus/BonusFactory.java:17]: Sounds a bit complex. Maybe: "The id of the source grading scale for the Bonus" ?

File: src/test/java/de/tum/in/www1/artemis/bonus/BonusFactory.java [hunk 2]
```diff
@@ -10,13 +10,13 @@
...
+     * @param bonusToGradingScaleId The id of the grading scale that is used as the target for the bonus
```
Comment (egekurt123) [inline src/test/java/de/tum/in/www1/artemis/bonus/BonusFactory.java:18]: the same here Maybe: "The id of the target grading scale for the Bonus" ?

**Review by egekurt123 — COMMENTED** [review]
lines 15-18, capitalizing "Bonus" would improve consistency in documentation

File: src/test/java/de/tum/in/www1/artemis/bonus/BonusFactory.java (hunk 1 above)
Comment (DominikRemo) [reply src/test/java/de/tum/in/www1/artemis/bonus/BonusFactory.java]: implemented

File: src/test/java/de/tum/in/www1/artemis/bonus/BonusFactory.java (hunk 2 above)
Comment (DominikRemo) [reply src/test/java/de/tum/in/www1/artemis/bonus/BonusFactory.java]: implemented, thank you for your review

**Review by JanaNF — APPROVED** [review]
Looks good to me

**Review by basak-akan — APPROVED** [review]
Javadoc changes looks good

**Review by LenaKahle — APPROVED** [review]
Looks good.

**Pull request merged**